vulnerabilities/importers/mozilla.py

```diff
@@ -0,0 +1,180 @@
+    for cve, advisory in data["advisories"].items():
+        # These may contain HTML tags
+        summary = BeautifulSoup(advisory.get("description", ""), features="lxml").get_text()
+
+        advisories.append(
+            Advisory(
+                summary=summary,
+                vulnerability_id=cve if is_cve(cve) else "",
+                impacted_package_urls=[],
+                resolved_package_urls=fixed_package_urls,
+                references=references,
+            )
+        )
+
+    return advisories
+
+
+def get_advisories_from_md(mfsa_id, lines) -> List[Advisory]:
+    yamltext, mdtext = split_markdown_front_matter(lines.read())
+    data = yaml.safe_load(yamltext)
+    data["mfsa_id"] = mfsa_id
+
+    fixed_package_urls = get_package_urls(data.get("fixed_in"))
+    references = get_yml_references(data)
+    cves = re.findall(r"CVE-\d+-\d+", yamltext + mdtext, re.IGNORECASE)
+    for cve in cves:
+        references.append(
+            Reference(
+                reference_id=cve,
+                url=f"https://cve.mitre.org/cgi-bin/cvename.cgi?name={cve}",
+            )
+        )
+
+    description = html_get_p_under_h3(markdown(mdtext), "description")
+
+    return [
+        Advisory(
+            summary=description,
+            vulnerability_id="",
+            impacted_package_urls=[],
+            resolved_package_urls=fixed_package_urls,
+            references=references,
+        )
+    ]
+
+
+def html_get_p_under_h3(html, h3: str):
+    soup = BeautifulSoup(html, features="lxml")
+    h3tag = soup.find("h3", text=lambda txt: txt.lower() == h3)
+    p = ""
+    if h3tag:
+        for tag in h3tag.next_siblings:
+            if tag.name:
+                if tag.name != "p":
+                    break
+                p += tag.get_text()
+    return p
+
+
+def mfsa_id_from_filename(filename):
+    match = MFSA_FILENAME_RE.search(filename)
+    if match:
+        return "mfsa" + match.group(1)
+
+    return None
+
+
+def get_package_urls(pkgs: List[str]) -> List[PackageURL]:
+    package_urls = [
+        PackageURL(
+            type="mozilla",
+            # pkg is of the form "Firefox ESR 1.21" or "Thunderbird 2.21"
+            name=pkg.rsplit(None, 1)[0],
+            version=pkg.rsplit(None, 1)[1],
+        )
+        for pkg in pkgs
+        if pkg
+    ]
+    return package_urls
+
+
+def get_yml_references(data: any) -> List[Reference]:
+    """
+    Returns a list of references
+    Currently only considers the given mfsa as a reference
+    """
+    # FIXME: Needs improvement
+    # Should we add 'bugs' section in references too?
+    # Should we add 'impact'/severity of CVE in references too?
+    # If yes, then fix alpine_linux importer as well
+    # Otherwise, do we need severity field for adversary as well?
+
+    severities = ["critical", "high", "medium", "low", "none"]
+    severity = "none"
+    if data.get("impact"):
+        impact = data.get("impact").lower()
+        for s in severities:
```

Comment:
I am not sure I get why you do things this way. What would be data examples?

Reply:
These are all unique impacts
```
mfsa2020-25:  high
mfsa2013-50:  Critical
mfsa2009-19:  High
mfsa2017-17:  critical
mfsa2009-15:  Low
mfsa2015-75:  Moderate
mfsa2020-07:  moderate
mfsa2007-06:  Critical (Firefox 2.0 not affected in default configuration)
mfsa2015-21:  Medium
mfsa2005-02:  Moderate (on a multiuser computer)
mfsa2005-28:  Critical (local)
mfsa2005-48:  Low (High for Mozilla Suite)
mfsa2005-59:  Severe
mfsa2010-55:  Low (Critical in Gecko 1.9.1 and earlier)
mfsa2005-10:  Moderate to Critical
```
I did that to make look like cvssv3.1_qr but now as we have generic_textual (https://github.com/nexB/vulnerablecode/pull/415) as well, perhaps I can just put the entire impact line over there.

Reply:
Just noticed that entries like `Critical (Firefox 2.0 not affected in default configuration)` exceed maximum value length.

Reply:
This is at best an artistic scale.... I would like to see what it looks like for recent values, say no less than 5 years old as this will weed out old things like FF 2.0
For the funky values I would rather have an explicit hardcoded mapping....and since they document it, we should IMHO create a Mozilla scoring system
@sbs2001 what's your take?

Reply:
Let's create a scoring system for mozilla using these values:
```
critical
severe
high
medium
moderate
low
``` 
in https://github.com/nexB/vulnerablecode/blob/main/vulnerabilities/severity_systems.py